packages/x-components/src/components/base-teleport.vue

```diff
@@ -38,21 +38,27 @@ export default defineComponent({
   },
   setup(props) {
     const teleportHost = document.createElement('div')
+    const observer = ref<MutationObserver | null>(null)
 
     watchEffect(() => {
       if (props.disabled) {
         teleportHost.remove()
         return
       }
-      teleportHost.className = `x-base-teleport x-base-teleport--${props.position}`
       const targetElement =
         typeof props.target === 'string' ? document.querySelector(props.target) : props.target
-      if (!targetElement) {
-        console.warn(`BaseTeleport: Target element "${props.target}" not found.`)
-        return
-      }
-      const position = props.position === 'onlychild' ? 'beforeend' : props.position
-      targetElement.insertAdjacentElement(position, teleportHost)
+
+      observer.value = new MutationObserver(() => {
+        teleportHost.className = `x-base-teleport x-base-teleport--${props.position}`
+
+        if (!targetElement) {
+          console.warn(`BaseTeleport: Target element "${props.target}" not found.`)
+          return
+        }
+        const position = props.position === 'onlychild' ? 'beforeend' : props.position
+        targetElement.insertAdjacentElement(position, teleportHost)
+      })
+      observer.value.observe(document.body, { childList: true, subtree: true })
```

Comment:
this is placed inside watchEffect, each time a ref changes, a new MutationObserver is created without remove the previous. (Memory leak & performance regresion)

I spect 2 non-simultaneous Observers that changes the value of `targetElement` now has to be a ref.
- If does not exist to know when exist (then remove the observer and create the other)
- If exist to know if its removed (then remove the observer and create the other)